File: ios/Curio/Curio/Platform/CurioNotifier.swift
//
//  CurioNotifier.swift
//  Curio
//
//  Ports: app/src/main/java/com/example/ui/CurioNotifier.kt
//
//  App-wide transient feedback channel. `BookmarkApp` registers `showMessage` so copy/share/settings
//  actions surface a glass-style toast instead of silent no-ops when the shell is active.
//

import Foundation

@MainActor
enum CurioNotifier {
    /// When set by the app shell, all `notify` calls route here instead of being dropped.
    static var showMessage: ((String) -> Void)?

    static func notify(_ message: String) {
        showMessage?(message)
    }
}

File: app/src/main/java/com/example/ui/CurioNotifier.kt
package com.example.ui

import android.content.Context
import android.widget.Toast

/**
 * App-wide transient feedback channel. [BookmarkApp] registers [showMessage] so copy/share/settings
 * actions can surface glass-style snackbars instead of system toasts when the shell is active.
 */
object CurioNotifier {
    var showMessage: ((String) -> Unit)? = null

    fun notify(context: Context, message: String) {
        val handler = showMessage
        if (handler != null) {
            handler(message)
        } else {
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
        }
    }
}
